### DISCUSSION TITLE
Help with a SQL query for a view collection of two counts

### DISCUSSION THREAD

**Discussion by jramiroz98** [pocketbase/pocketbase#4059] [Q&A]
Hello there! I'm working on a table to view the number of visits on my page. It works like this:
`SELECT id,count(my_visits.id) as visit_count from my_visits`

But now I would like to add the count of users. Can't find a way to do this. If anyone knows a query to try out or documentation to read I would appreciate it a lot.
Been trying the following but no success:

```
SELECT
    'my_visits' AS table_name,
    COUNT(id) AS entry_count
FROM my_visits

UNION ALL

SELECT
    'users' AS table_name,
    COUNT(id) AS entry_count
FROM users;
```

  Thanks in advance! and have an awesome day!

**Comment by ganigeorgiev** [comment]
How the visits are related to your users? Or in other words - what is the structure of the 2 tables/collections?

If you have a relation field between the 2 tables you can `JOIN` them and use `GROUP BY users.id` to group the count aggregation per each user.

**Reply by jramiroz98** [reply]
To be honest they dont have a relation. One table has users, the 'my_visits' table only creates an entry every time the homepage loads. I just wanted a quick and easy layout to view both

Right now this sort of worked but in a weird way haha
```

SELECT id, visit_count 
FROM (
    SELECT id,count(my_visits.id) as visit_count from my_visits
    UNION
    SELECT id,count(users.id) as users_count from users
)
```

I have 36 visits and 1 user:
![image](https://github.com/pocketbase/pocketbase/assets/80351867/8fa375eb-60f6-4d5d-9494-acbed5fe1cf8)

**Reply by ganigeorgiev** [reply]
So you just want a view that returns the number of records from the 2 collections? If that's the case, then you can try something like:

```sql
SELECT id, totalCount 
FROM (
    SELECT 'my_visits' as id, count(*) as totalCount from my_visits
    UNION
    SELECT 'users' as id, count(*) as totalCount from users
)
```
The above should create a View collection with 2 fields - the name of the count table as "id" and the count as "totalCount".

**Reply by jramiroz98** [reply]
Awesome man!! Thanks!
Works like a charm
Would there be a way to make both show up in just one record?

**Reply by ganigeorgiev** [reply]
You can try with something like:

```sql
SELECT
  1 as id,
  (select count(*) from my_visits) as visitsCount,
  (select count(*) from users) as usersCount

```
